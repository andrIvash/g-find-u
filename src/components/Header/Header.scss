@import '../../styles/abstracts/functions';
@import '../../styles/abstracts/breakpoints';

.header {
    display: flex;
    align-items: center;
    padding: rem(0, 10);
    margin-bottom: rem(20);
}

.header__logo {
    display: flex;
    align-items: center;
    margin-right: rem(10);
}

.header__logoIcon.icon {
    width: rem(30);
    height: rem(30);
}

@media screen {
    @include break-min-max(rem(425), rem(1100)) {
        .header__logo {
            margin-right: rem(10);
        }

        .header__logoIcon.icon {
            width: rem(20);
            height: rem(20);
        }
    }

    @include break-min-max(rem(320), rem(500)) {
        .header__logo {
            margin-right: rem(5);
        }

        .header__logoIcon.icon {
            width: rem(15);
            height: rem(15);
        }
    }
}