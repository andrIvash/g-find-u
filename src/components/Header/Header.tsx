import React, { useContext } from 'react';
import { DarkThemeContext } from '../../context/darkThemeContext';

import './Header.scss';

export const Header = () => {
    const darkTheme = useContext(DarkThemeContext);
    return (
        <div className={`header ${darkTheme ? "header--dark" : ""}`}>
            <h2>Header</h2>
            <p>
                test
            </p>
        </div>
    );
};