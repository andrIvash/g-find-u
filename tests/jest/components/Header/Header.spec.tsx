import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../../../../src/components/Header';

describe('Header Component', () => {
    it('renders without crashing', () => {
        render(<Header />);
        expect(screen.getByText('GitHub Finder')).toBeInTheDocument();
    });

    it('displays the logo image', () => {
        render(<Header />);
        const logoImg = screen.getByRole('img');
        expect(logoImg).toHaveClass('icon rotate header__logoIcon');
    });

    it('has correct title text', () => {
        render(<Header />);
        const titleElement = screen.getByText('GitHub Finder');
        expect(titleElement).toBeInTheDocument();
    });

    it('has the correct class for the header', () => {
        render(<Header />);
        const headerElement = screen.getByText('GitHub Finder').parentNode;
        expect(headerElement).toHaveClass('header');
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
    });
});