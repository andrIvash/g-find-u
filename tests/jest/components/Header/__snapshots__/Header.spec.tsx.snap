// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="header__logo"
    >
      <img
        class="icon rotate header__logoIcon"
        src=""
      />
    </div>
    <h1>
      GitHub Finder
    </h1>
  </div>
</DocumentFragment>
`;
